### PR TITLE
Dev merge

### PR DESCRIPTION
**Pull request by hlky**

### CODE REVIEW

File: frontend/frontend.py [hunk 1]
```diff
@@ -158,30 +158,30 @@ def draw_gradio_ui(opt, img2img=lambda x: x, txt2img=lambda x: x,imgproc=lambda
                 with gr.Row().style(equal_height=False):
                     with gr.Column():
                         gr.Markdown('#### Img2Img Input')
-                        img2img_image_editor = gr.Image(value=sample_img2img, source="upload", interactive=True,
```
Comment (hlky) [inline frontend/frontend.py:161]: Conflict with img2img fixes?

File: frontend/frontend.py [hunk 2]
```diff
@@ -158,30 +158,30 @@ def draw_gradio_ui(opt, img2img=lambda x: x, txt2img=lambda x: x,imgproc=lambda
...
+                        img2img_image_editor = gr.Image(value=sample_img2img, source="upload", interactive=False,
+                                                      type="pil", tool="sketch", visible=False, image_mode="RGBA",
                                                       elem_id="img2img_mask")
 
                         with gr.Tabs():
                             with gr.TabItem("Editor Options"):
                                 with gr.Row():
                                     img2img_image_editor_mode = gr.Radio(choices=["Mask", "Crop", "Uncrop"], label="Image Editor Mode",
-                                                             value="Crop", elem_id='edit_mode_select')
+                                                             value="Mask", elem_id='edit_mode_select', visible=False)
```
Comment (hlky) [inline frontend/frontend.py:172]: Conflict?

File: frontend/frontend.py [hunk 3]
```diff
@@ -158,30 +158,30 @@ def draw_gradio_ui(opt, img2img=lambda x: x, txt2img=lambda x: x,imgproc=lambda
...
-                                                value=img2img_mask_modes[img2img_defaults['mask_mode']], visible=False)
+                                                value=img2img_mask_modes[img2img_defaults['mask_mode']], visible=True)
 
                                     img2img_mask_blur_strength = gr.Slider(minimum=1, maximum=10, step=1,
                                                                label="How much blurry should the mask be? (to avoid hard edges)",
                                                                value=3, visible=False)
 
                                     img2img_resize = gr.Radio(label="Resize mode",
-                                                choices=["Just resize"],
+                                                choices=["Just resize", "Crop and resize", "Resize and fill"],
```
Comment (hlky) [inline frontend/frontend.py:182]: Conflict?

File: frontend/frontend.py [hunk 4]
```diff
@@ -158,30 +158,30 @@ def draw_gradio_ui(opt, img2img=lambda x: x, txt2img=lambda x: x,imgproc=lambda
...
                         with gr.Tabs():
                             with gr.TabItem("Editor Options"):
                                 with gr.Row():
                                     img2img_image_editor_mode = gr.Radio(choices=["Mask", "Crop", "Uncrop"], label="Image Editor Mode",
-                                                             value="Crop", elem_id='edit_mode_select')
+                                                             value="Mask", elem_id='edit_mode_select', visible=False)
                                     img2img_mask = gr.Radio(choices=["Keep masked area", "Regenerate only masked area"],
                                                 label="Mask Mode", type="index",
-                                                value=img2img_mask_modes[img2img_defaults['mask_mode']], visible=False)
+                                                value=img2img_mask_modes[img2img_defaults['mask_mode']], visible=True)
```
Comment (hlky) [inline frontend/frontend.py:175]: Conflict?

File: frontend/frontend.py [hunk 5]
```diff
@@ -158,30 +158,30 @@ def draw_gradio_ui(opt, img2img=lambda x: x, txt2img=lambda x: x,imgproc=lambda
...
                                     img2img_mask_blur_strength = gr.Slider(minimum=1, maximum=10, step=1,
                                                                label="How much blurry should the mask be? (to avoid hard edges)",
                                                                value=3, visible=False)
 
                                     img2img_resize = gr.Radio(label="Resize mode",
-                                                choices=["Just resize"],
+                                                choices=["Just resize", "Crop and resize", "Resize and fill"],
                                                 type="index",
-                                                value=img2img_resize_modes[img2img_defaults['resize_mode']])
+                                                value=img2img_resize_modes[img2img_defaults['resize_mode']], visible=False)
```
Comment (hlky) [inline frontend/frontend.py:184]: Conflict?

File: frontend/frontend.py [hunk 6]
```diff
@@ -256,16 +256,22 @@ def draw_gradio_ui(opt, img2img=lambda x: x, txt2img=lambda x: x,imgproc=lambda
 
                 img2img_image_editor_mode.change(
                     uifn.change_image_editor_mode,
-                    [img2img_image_editor_mode, img2img_image_editor, img2img_resize, img2img_width, img2img_height],
+                    [img2img_image_editor_mode,
```
Comment (hlky) [inline frontend/frontend.py:259]: Conflict?

File: frontend/frontend.py [hunk 7]
```diff
@@ -299,11 +305,11 @@ def draw_gradio_ui(opt, img2img=lambda x: x, txt2img=lambda x: x,imgproc=lambda
                                                            )
 
                 img2img_func = img2img
-                img2img_inputs = [img2img_prompt, img2img_image_editor_mode, img2img_image_editor, img2img_image_mask, img2img_mask,
+                img2img_inputs = [img2img_prompt, img2img_image_editor_mode, img2img_mask,
```
Comment (hlky) [inline frontend/frontend.py:308]: Conflict?

File: frontend/ui_functions.py [hunk 8]
```diff
@@ -6,14 +6,17 @@
 import re
 
 
-def change_image_editor_mode(choice, cropped_image, resize_mode, width, height):
+def change_image_editor_mode(choice, cropped_image, masked_image, resize_mode, width, height):
```
Comment (hlky) [inline frontend/ui_functions.py:9]: Conflict?

**Comment by codedealer** [comment]
Oh my god, I was doing this at the moment at was about to give up 😭

You're a super being!

File: frontend/frontend.py (hunk 1 above)
Comment (ZeroCool940711) [reply frontend/frontend.py:161]: I think this was a fix for the img2img as you said, this part is not on the dev branch version of this file, might have to be added manually again.

File: frontend/frontend.py (hunk 3 above)
Comment (ZeroCool940711) [reply frontend/frontend.py:182]: I think the other options are not needed anymore and "Just resize" is what should be used, I remember you said the other options were not useful.

File: scripts/webui.py [hunk 9]
```diff
@@ -1290,22 +1299,22 @@ def img2img(prompt: str, image_editor_mode: str, init_info: any, init_info_mask:
...
-        #image = image.convert("RGB") #todo: mask mode -> ValueError: could not convert string to float:
+        #image = image.convert("RGB")
         image = np.array(image).astype(np.float32) / 255.0
         image = image[None].transpose(0, 3, 1, 2)
         image = torch.from_numpy(image)
 
         mask_channel = None
         if image_editor_mode == "Uncrop":
-            alpha = init_img.convert("RGB")
+            alpha = init_img.convert("RGBA")
```
[the remark this answers is not in the document]
Comment (ZeroCool940711) [reply scripts/webui.py:1309]: I think RGBA would allow for alpha channel to work without issues, not 100% sure.

**Review by codedealer — APPROVED** [review]

**Pull request merged**